src/components/EditPostModal/EditPostModal.jsx

```diff
@@ -0,0 +1,132 @@
+import React, { useEffect, useState } from "react";
+import { useDispatch } from "react-redux";
+import { useNavigate } from "react-router-dom";
+import { FileUploader } from "react-drag-drop-files";
+import { editPost } from "redux/reducers/postsSlice";
+
+const fileTypes = ["JPEG", "PNG", "GIF"];
+
+const EditPostModal = ({
+  postID,
+  caption: initialCaption,
+  image: initialImageUrl,
+  setShowEditPostModal,
+}) => {
+  const [caption, setCaption] = useState(initialCaption);
+  const [selectedImage, setSelectedImage] = useState();
+  const [imgUrl, setImgUrl] = useState(initialImageUrl);
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    setCaption(initialCaption);
+    setImgUrl(initialImageUrl);
+  }, [initialCaption, initialImageUrl]);
+
+  const fileChangeHandler = (selectedImage) => {
+    setSelectedImage(selectedImage);
+  };
+
+  const uploadImage = async (e) => {
+    e.preventDefault();
+    const formData = new FormData();
+    formData.append("file", selectedImage);
+    formData.append("upload_preset", process.env.REACT_APP_UPLOAD_PRESET);
+
+    fetch("https://api.cloudinary.com/v1_1/dl0nhw7w3/image/upload", {
+      method: "POST",
+      body: formData,
+    })
+      .then((response) => {
+        return response.json();
```

Comment:
Suggestion: Since you're using async function, you can use await here instead of then callbacks